ui/src/utils/queryTransitions.js

```diff
@@ -119,10 +130,16 @@ export function applyFuncsToField(
     time: shouldRemoveFuncs ? null : defaultGroupBy,
   })
 
-  return Object.assign({}, query, {
-    fields: nextFields,
-    groupBy: nextGroupBy,
-  })
+  const nextQuery = {...query, fields: nextFields, groupBy: nextGroupBy}
+
+  // fill is not valid for kapacitor query configs since there is no actual
+  // query and all alert rules create stream-based tasks currently
+  if (!isKapacitorRule) {
+    const nextFill = NULL_STRING
+    Object.assign(nextQuery, {fill: nextFill})
+  }
```

Comment:
Use spread please.  But also, without a new empty object as the first argument line 139 `Object.assign` is modifying `nextQuery` in place.

Reply:
It's intended to modify nextQuery in place

Reply:
Right.  Def don't do that plz :)